provider/aws/aws.go

```diff
@@ -172,8 +172,13 @@ func NewRateLimitedRoute53API(route53api Route53API, rateLimit int) *RateLimited
 
 func (r *RateLimitedRoute53API) ListResourceRecordSetsPagesWithContext(ctx context.Context, input *route53.ListResourceRecordSetsInput, fn func(resp *route53.ListResourceRecordSetsOutput, lastPage bool) (shouldContinue bool), opts ...request.Option) error {
 	r.rateLimiter.Take()
-	route53RequestsTotal.Inc()
-	err := r.client.ListResourceRecordSetsPagesWithContext(ctx, input, fn, opts...)
+	f := func(resp *route53.ListResourceRecordSetsOutput, lastPage bool) (shouldContinue bool) {
```

Comment:
perhaps instead of 🅵 this could be like `fnWithThrottling`?

Reply:
Sure